MailerQ/ApiClient/ApiClient.cs

```
@@ -15,186 +17,200 @@ public class ApiClient : IApiClient
-                return JsonConvert.DeserializeObject<List<Pause>>(result);
+                var result = MailerQApiClientGetAsync(pause.Endpoint);
+                return JsonConvert.DeserializeObject<List<Pause>>(result.Result);
             }
             catch (Exception exception)
             {
                 _logger.LogError(exception,
                     "Endpoint: {Endpoint}. Method: {Method}.",
-                    EndpointPause,
-                    MethodGet);
+                    pause.Endpoint,
+                    nameof(Get));
                 throw exception;
             }
         }
 
-        private string MailerQApiClientGet(string endpoint, string method)
+        private async Task<string> MailerQApiClientGetAsync(string endpoint)
         {
             _apiUri.Path = Path.Combine(_apiUri.Path, endpoint);
-            var request = (HttpWebRequest)WebRequest.Create(_apiUri.Uri);
 
-            request.Method = method;
-            request.Headers.Add("Content-Type", "application/json");
-            request.Headers.Add("Authorization", "Bearer " + _apiClientSettings.Token);
+            var client = _httpFactory.CreateClient();
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _apiClientSettings.Token);
+            client.DefaultRequestHeaders.Add("Accept", "application/json");
 
-            var content = string.Empty;
-
-            using (var response = (HttpWebResponse)request.GetResponse())
+            var httpResponse = await client.GetAsync(_apiUri.Uri);
+            if (httpResponse.StatusCode == HttpStatusCode.OK)
             {
-                using var stream = response.GetResponseStream();
-                using var streamReader = new StreamReader(stream);
-                content = streamReader.ReadToEnd();
+                return await httpResponse.Content.ReadAsStringAsync();
             }
-            return content;
+
+            return null;
```

I think that it is not a good idea hiding errors returning null.